sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/EmptyNestedBlock.cs

```diff
@@ -0,0 +1,313 @@
+                },
+                SyntaxKind.SimpleDoLoopBlock,
+                SyntaxKind.DoLoopUntilBlock,
+                SyntaxKind.DoLoopWhileBlock,
+                SyntaxKind.DoUntilLoopBlock,
+                SyntaxKind.DoWhileLoopBlock,
+                SyntaxKind.ForBlock,
+                SyntaxKind.ForEachBlock,
+                // The Else and ElseIf blocks are inside the MultiLineIfBlock
+                SyntaxKind.MultiLineIfBlock,
+                SyntaxKind.SelectBlock,
+                // The CatchBlock and FinallyBlock are inside the TryBlock
+                SyntaxKind.TryBlock,
+                SyntaxKind.UsingBlock,
+                SyntaxKind.WhileBlock,
+                SyntaxKind.WithBlock);
+        }
+
+        /**
+         * Verify that the given block has no statements and no comments inside
+         * (notable exception: the Select block)
+         *
+         * Note: Roslyn maps the comments which are inside a block as trivia of the following block
+         * e.g. in the below snippet, the comment will be part of the Finally block
+         *
+         * Try
+         *   ' my comment
+         * Finally
+         * End Try
+         */
+
+        private void VerifyEmptyBlocks(SyntaxNode node)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.SimpleDoLoopBlock:
+                case SyntaxKind.DoLoopUntilBlock:
+                case SyntaxKind.DoLoopWhileBlock:
+                case SyntaxKind.DoUntilLoopBlock:
+                case SyntaxKind.DoWhileLoopBlock:
+                    VisitDoLoopBlock((DoLoopBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.ForBlock:
+                    VisitForBlock((ForBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.ForEachBlock:
+                    VisitForEachBlock((ForEachBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.MultiLineIfBlock:
+                    VisitMultiLineIfBlock((MultiLineIfBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.SelectBlock:
+                    VisitSelectBlock((SelectBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.TryBlock:
+                    VisitTryBlock((TryBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.UsingBlock:
+                    VisitUsingBlock((UsingBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.WhileBlock:
+                    VisitWhileBlock((WhileBlockSyntax)node);
+                    break;
+
+                case SyntaxKind.WithBlock:
+                    VisitWithBlock((WithBlockSyntax)node);
+                    break;
+
+                default:
+                    throw new InvalidOperationException($"Did not expect this syntax kind: {node.Kind()}");
```

Comment:
I can understand the throw but if the language evolves that means that our rule is going to start failing.